lib/diaspora_federation/entities/relayable_retraction.rb

```diff
@@ -1,10 +1,33 @@
 module DiasporaFederation
   module Entities
+    # this entity represents a claim of deletion of a previously federated relayable entity
+    #
+    # @see Validators::RelayableRetractionValidator
     class RelayableRetraction < Entity
+      # @!attribute [r] parent_author_signature
+      #   Contains a signature of the entity using the private key of the author of a parent post
+      #   @return [String] parent author signature
       property :parent_author_signature
+
+      # @!attribute [r] target_guid
+      #   guid of a post to be deleted
+      #   @see HCard#guid
+      #   @return [String] target guid
       property :target_guid
+
+      # @!attribute [r] target_type
+      #   @return [String] target type
       property :target_type
+
+      # @!attribute [r] sender_id
+      #   The diaspora ID of the person who deletes a post
+      #   @see Person#diaspora_id
+      #   @return [String] diaspora ID
       property :sender_id, xml_name: :sender_handle
+
+      # @!attribute [r] target_author_signature
+      #   Contains a signature of the entity using the private key of the author of a post itself
```

Comment:
Yes, it is the signature of the relayable, I must reword it.

It works pretty much the same way the relayable federation works.

When author of the parent post deletes the comment this signature [is not checked](https://github.com/diaspora/diaspora/blob/develop/lib/diaspora/federated/relayable_retraction.rb#L51-L53).